src/OpenTelemetry.Exporter.Prometheus/Implementation/PrometheusSerializerExt.cs

```diff
@@ -66,19 +66,25 @@ public static int WriteMetric(byte[] buffer, int cursor, Metric metric)
 
                     // Counter and Gauge
                     cursor = WriteMetricName(buffer, cursor, metric.Name, metric.Unit);
-                    buffer[cursor++] = unchecked((byte)'{');
 
-                    for (var i = 0; i < keys.Length; i++)
+                    int numberOfKeys = keys?.Length ?? 0;
+                    if (numberOfKeys > 0)
```

Comment:
@reyang I think this was a bug. We were getting nullref exception when keys/labels were not specified.

Reply:
Yeah, I noticed this.

Reply:
Let me add a test case for it once this PR is merged.

Reply:
#2612